src/aiida/orm/nodes/data/remote/base.py

```diff
@@ -96,22 +93,15 @@ def listdir(self, relpath='.'):
         authinfo = self.get_authinfo()
 
         with authinfo.get_transport() as transport:
-            try:
-                full_path = os.path.join(self.get_remote_path(), relpath)
-                transport.chdir(full_path)
-            except OSError as exception:
-                if exception.errno in (2, 20):  # directory not existing or not a directory
-                    exc = OSError(
-                        f'The required remote folder {full_path} on {self.computer.label} does not exist, is not a '
-                        'directory or has been deleted.'
-                    )
-                    exc.errno = exception.errno
-                    raise exc from exception
-                else:
-                    raise
+            full_path = os.path.join(self.get_remote_path(), relpath)
```

Comment:
Might be a scope creep here, and probably unnecesary for the changes of this PR, but we could, at some point, fully remove
`os.path` calls with `pathlib` as it's now the recommended way to handle file system paths.

Reply:
Agreed!